[Fixed Position] Modify fixed position to use the Selection API

psarram opened the pull request:
Fixes #1848

**Author Checklist**

- Changes address original issue? Y
- Unit tests included and/or updated with changes? Not yet
- Command line build passes? Y
- Changes have been smoke-tested? Y

@VWoeltjen @larkin I'll have to update the unit tests but the changes are ready for review. 
@charlesh88 Elements can now be selected in browse mode and so some style updates might be needed.

VWoeltjen commented on platform/features/layout/src/FixedController.js:
```diff
@@ -492,42 +505,47 @@ define(
-            var selection = this.selection;
-            return selection && ((arguments.length > 0) ?
-                    selection.selected(element) : selection.get());
+        FixedController.prototype.selected = function (elementProxy) {
```
Given the changes to functionality (mentioned in JSDoc comment) I'd change the method name to `shouldSelect`

psarram commented:
**Author Checklist**

- Changes address original issue? Y
- Unit tests included and/or updated with changes? Y
- Command line build passes? Y
- Changes have been smoke-tested? Y

@VWoeltjen I addressed the requested change and fixed the tests.

VWoeltjen commented:
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
